vault/expiration.go

```diff
@@ -2291,6 +2345,47 @@ func (m *ExpirationManager) walkLeases(walkFn leaseWalkFunction) error {
 	return nil
 }
 
+// must be called with m.pendingLock held
+func (m *ExpirationManager) removeFromPending(ctx context.Context, leaseID string) {
+	if info, ok := m.pending.Load(leaseID); ok {
+		pending := info.(pendingInfo)
+		pending.timer.Stop()
+		m.pending.Delete(leaseID)
+		m.leaseCount--
+		// Log but do not fail; unit tests (and maybe Tidy on production systems)
+		if err := m.core.quotasHandleLeases(ctx, quotas.LeaseActionDeleted, []string{leaseID}); err != nil {
+			m.logger.Error("failed to update quota on revocation", "error", err)
+		}
+	}
+}
+
+// note: must be called with pending lock held
+func (m *ExpirationManager) markLeaseAsZombie(ctx context.Context, le *leaseEntry, err error) {
+	if le == nil {
+		m.logger.Warn("attempted to mark nil lease as zombie")
+		return
+	}
+	if le.isZombie() {
+		m.logger.Info("attempted to re-mark lease as zombie", "original_error", le.RevokeErr, "new_error", err.Error())
+		return
+	}
+
+	errStr := err.Error()
+	if len(errStr) == 0 {
+		errStr = "no error message given"
+	}
+	if len(errStr) > maxZombieErrorLength {
```

Comment:
This seems like an interesting case to log as well. What do you think?

Reply:
hmm, if you feel strongly that it should be logged i can put a log in. i don't think it's that important though - this is mostly to avoid over-consuming storage/memory, and i think 240 characters is enough to get a sense of what's going on

Reply:
That's fine, definitely don't feel strongly about it. Feel free to mark this one as resolved!